lib/user-interface/react-app/src/components/chatbot/multi-chat.tsx

```diff
@@ -0,0 +1,292 @@
+import { useContext, useEffect, useState } from "react";
+import {
+  SelectProps,
+  SpaceBetween,
+  Button,
+  Select,
+  ColumnLayout,
+  Grid,
+} from "@cloudscape-design/components";
+import { v4 as uuidv4 } from "uuid";
+import { AppContext } from "../../common/app-context";
+import { ApiClient } from "../../common/api-client/api-client";
+import ChatMessage from "./chat-message";
+import MultiChatInputPanel, { ChatScrollState } from "./multi-chat-input-panel";
+import styles from "../../styles/chat.module.scss";
+import useWebSocket, { ReadyState } from "react-use-websocket";
+import { OptionsHelper } from "../../common/helpers/options-helper";
+import { Auth } from "aws-amplify";
+import {
+  ChatBotConfiguration,
+  ChatBotAction,
+  ChatBotHistoryItem,
+  ChatBotMessageResponse,
+  ChatBotMessageType,
+  ChatBotRunRequest,
+  ChatBotMode,
+  ChabotInputModality,
+  ChabotOutputModality,
+} from "./types";
+
+import { ApiResult, ModelItem, WorkspaceItem, ResultValue } from "../../common/types";
+import { getSelectedModelMetadata, updateChatSessions } from "./utils";
+import LLMConfigDialog from "./llm-config-dialog";
+
+export interface ChatSession {
+  configuration: ChatBotConfiguration;
+  model?: SelectProps.Option;
+  modelMetadata?: ModelItem;
+  id: string;
+  loading: boolean;
+  running: boolean;
+  messageHistory: ChatBotHistoryItem[];
+}
+
+function createNewSession(): ChatSession {
+  return {
+    id: uuidv4(),
+    loading: false,
+    running: false,
+    messageHistory: [],
+    configuration: {
+      files: [],
+      streaming: true,
+      showMetadata: false,
+      maxTokens: 512,
+      temperature: 0.1,
+      topP: 0.9,
+    },
+  }
+}
+
+export default function MultiChat() {
+  const appContext = useContext(AppContext);
+  const [socketUrl, setSocketUrl] = useState<string | null>(null);
+  const [chatSessions, setChatSessions] = useState<ChatSession[]>([]);
+  const [models, setModels] = useState<ModelItem[]>([]);
+  const [workspaces, setWorkspaces] = useState<WorkspaceItem[]>([]);
+  const [enableAddModels, setEnableAddModels] = useState(true);
+  const [llmToConfig, setLlmToConfig] = useState<ChatSession | undefined>(
+    undefined
+  );
+  const [showMetadata, setShowMetadata] = useState(false);
+
+  const { sendJsonMessage, readyState } = useWebSocket(socketUrl, {
+    share: true,
+    shouldReconnect: () => true,
```

Comment:
should we include [heartbeat](https://github.com/aws-samples/aws-genai-llm-chatbot/blob/f3a7ad2c0f95e6b5c7d6b03fe35de7d778ea8c74/lib/user-interface/react-app/src/components/chatbot/chat-input-panel.tsx#L104) `onOpen` here too?

Reply:
Yes